the-collab-lab/tcl-33-smart-shopping-list
# Is jt new list

## Description



This code:

- Generates a new, unique token
- Saves the token to localStorage
- Shows the user the list view (redirects the user to the List View page)

## Related Issue


Closes #3

## Acceptance Criteria


- @the-collab-lab/shopping-list-utils has been added as a dependency to the project.
- For users who do not already have a token or a list, a button or link exists on the home screen that allows them to create a new list
- Clicking the button/link generates a new token and saves it to localStorage
- Once the token has been created and saved, the user is redirected to the "list" view
- For users who do have a token already saved in localStorage, they should be shown/redirected to the "list" view

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|   | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before



![Homepage Before)](https://user-images.githubusercontent.com/50176611/137527451-1f0599e0-f070-45c0-ab54-08da89cde60f.png)


### After



[Click here for video link to added feature](https://www.loom.com/share/cd81ea4fb89e44b9a900db0c5ba6af1e).

![Screenshot (376)](https://user-images.githubusercontent.com/50176611/137527715-5bfcef0a-0f29-4495-b0fc-57911a0aeeb0.png)
![Screenshot (379)](https://user-images.githubusercontent.com/50176611/137528019-7f64d73d-c844-434d-8166-4d8e402128bd.png)


## Testing Steps / QA Criteria



- Open the file and your terminal then enter "npm start" to automagically open the file in localhost:3000
- Check local storage to confirm it is empty. If it is not empty, please clear it.
- Click on the "Create Shopping List" button. It should:
1. Create a key "Token" and value equal of three words (they may be nonsense words) in local storage
2. Redirect you (the user) to the List View page.
3. Make it impossible to redirect to the Home page using the navigation bar.
